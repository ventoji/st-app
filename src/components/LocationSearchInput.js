/**
 * LocationSearchInput is a component from the third party library
 * https://www.npmjs.com/package/react-places-autocomplete
 * handleSelect and handleChange were customized for the app to handle
 * state redux.
 */
import React from 'react';
import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng,
} from 'react-places-autocomplete';

class LocationSearchInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      address: '',
      error: '',
    };
  }

  handleChange = (address) => {
    this.setState({ address });
    if (this.props.onSearchChange) this.props.onSearchChange({ address });
  };

  handleSelect = (address) => {
    geocodeByAddress(address)
      .then((results) => getLatLng(results[0]))
      .then((latLng) => {
        console.log('Success', latLng);
        if (this.props.onSelectChange)
          this.props.onSelectChange({ address, latLng });
      })
      .catch((error) => {
        this.setState({
          error: 'Connecting to Google Maps Api Failed',
        });
      });
  };

  render() {
    return (
      <PlacesAutocomplete
        value={this.state.address}
        onChange={this.handleChange}
        onSelect={this.handleSelect}
      >
        {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
          <div>
            <input
              {...getInputProps({
                placeholder: 'Search Places ...',
                className: 'location-search-input',
              })}
            />
            <div className="autocomplete-dropdown-container">
              {loading && <div>Loading...</div>}
              {suggestions.map((suggestion) => {
                const className = suggestion.active
                  ? 'suggestion-item--active'
                  : 'suggestion-item';
                // inline style for demonstration purpose
                const style = suggestion.active
                  ? {
                      backgroundColor: '#fafafa',
                      cursor: 'pointer',
                      fontWeight: 'bold',
                    }
                  : { backgroundColor: '#ffffff', cursor: 'pointer' };
                return (
                  <div
                    {...getSuggestionItemProps(suggestion, {
                      className,
                      style,
                    })}
                  >
                    <span>{suggestion.description}</span>
                  </div>
                );
              })}
            </div>
          </div>
        )}
      </PlacesAutocomplete>
    );
  }
}

export default LocationSearchInput;
