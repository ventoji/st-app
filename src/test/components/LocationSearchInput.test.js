import React from 'react';
import { shallow } from 'enzyme';
import LocationSearchInput from '../../components/LocationSearchInput';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<LocationSearchInput />);
});

test('should render LocationSearchInput correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
