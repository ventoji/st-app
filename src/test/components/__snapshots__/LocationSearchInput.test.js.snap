// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LocationSearchInput correctly 1`] = `
<PlacesAutocomplete
  debounce={200}
  highlightFirstSuggestion={false}
  onChange={[Function]}
  onError={[Function]}
  onSelect={[Function]}
  searchOptions={Object {}}
  shouldFetchSuggestions={true}
  value=""
>
  <Component />
</PlacesAutocomplete>
`;
